### PR TITLE
Feat/8.2 xcp ng drivers

### PR DESCRIPTION
**Pull request by Wescoeur**
To test:
- [ ] CephFS
- [ ] GlusterFS
- [ ] XFS
- [ ] ZFS

See: https://github.com/xcp-ng/xcp/issues/411

### CODE REVIEW

File: drivers/CephFSSR.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+import errno
+import os
+import syslog as _syslog
+import xmlrpclib
+from syslog import syslog
+
+# careful with the import order here
+# FileSR has a circular dependency:
+# FileSR- > blktap2 -> lvutil -> EXTSR -> FileSR
```
Comment (stormi) [inline drivers/CephFSSR.py]: Nitpicking: `FileSR- >` => `FileSR ->`

File: drivers/CephFSSR.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+CAPABILITIES = ["SR_PROBE", "SR_UPDATE", "SR_CACHING",
+                "VDI_CREATE", "VDI_DELETE", "VDI_ATTACH", "VDI_DETACH",
+                "VDI_UPDATE", "VDI_CLONE", "VDI_SNAPSHOT", "VDI_RESIZE", "VDI_MIRROR",
+                "VDI_GENERATE_CONFIG",
+                "VDI_RESET_ON_BOOT/2", "ATOMIC_PAUSE"]
+
+CONFIGURATION = [['server', 'Ceph server(s) (required, ex: "192.168.0.12" or "10.10.10.10,10.10.10.26")'],
+                 ['serverpath', 'Ceph FS path (required, ex: "/")'],
+                 ['serverport', 'ex: 6789'],
+		 ['options', 'Ceph FS client name, and secretfile (required, ex: "name=admin,secretfile=/etc/ceph/admin.secret")']
```
Comment (stormi) [inline drivers/CephFSSR.py]: indentation

File: drivers/GlusterFSSR.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+CONFIGURATION = [['server', 'Full path to share on gluster server (required, ex: "192.168.0.12:/gv0")'],
+                 ['backupservers', 'list of servers separated by ":"'],
+                 ['fetchattempts', 'number of attempts to fetch files before switching to the backup server']
+                 ]
+
+DRIVER_INFO = {
+    'name': 'GlusterFS VHD',
+    'description': 'SR plugin which stores disks as VHD files on a GlusterFS storage',
+    'vendor': 'Vates SAS',
+    'copyright': '(C) 2017 Vates SAS',
```
Comment (stormi) [inline drivers/GlusterFSSR.py]: Passons le à 2020 puisque le driver de 2017 existait sous un autre nom

File: drivers/XFSSR.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        if not self.dconf.has_key('device') or not self.dconf['device']:
+            raise xs_errors.XenError('ConfigDeviceMissing')
+
+        self.root = self.dconf['device']
+        for dev in self.root.split(','):
+            if not self._isvalidpathstring(dev):
+                raise xs_errors.XenError('ConfigDeviceInvalid', \
+                      opterr='path is %s' % dev)
+        self.path = os.path.join(SR.MOUNT_BASE, sr_uuid)
+        self.vgname = EXT_PREFIX + sr_uuid
```
Comment (stormi) [inline drivers/XFSSR.py:79]: This looks wrong but it's not from you and is too late (users already have created SRs with the experimental driver)

File: drivers/XFSSR.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+            return EXTFileVDI(self, uuid)
+        return EXTFileVDI(self, uuid)
+
+    @staticmethod
+    def _is_xfs_available():
+        import distutils.spawn
+        return distutils.spawn.find_executable('mkfs.xfs')
+
+
+class EXTFileVDI(FileSR.FileVDI):
```
Comment (stormi) [inline drivers/XFSSR.py]: I know it's not from you, but that class name looks wrong. If we can fix it without causing issues for existing users, let's fix it.

File: drivers/XFSSR.py (hunk 4 above)
Comment (Wescoeur) [reply drivers/XFSSR.py:79]: Yes, I suppose we can open a low-priority issue concerning this naming.

File: drivers/XFSSR.py (hunk 5 above)
Comment (Wescoeur) [reply drivers/XFSSR.py]: It's safe. I modify that.

**Pull request merged**